test/e2e/nbthread_test.go

```diff
@@ -0,0 +1,87 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const nbthreadRouterDeployTimeout time.Duration = 1 * time.Minute
+
+// NOTE: This test will mutate the default ingresscontroller
+func TestRouteNbthreadIngressController(t *testing.T) {
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	routerDeployment, err := getDeployment(t, kclient, controller.RouterDeploymentName(ic), 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get router deployment: %v", err)
+	}
+
+	defaultThreadsStr := strconv.Itoa(ingress.RouterHAProxyThreadsDefaultValue)
+
+	// by default, the router deployment should have ROUTER_THREADS set to 4
+	if err := waitForDeploymentEnvVar(t, kclient, routerDeployment, nbthreadRouterDeployTimeout, "ROUTER_THREADS", defaultThreadsStr); err != nil {
+		t.Fatalf("expected router deployment to have spec.threading.count unset: %v", err)
```

Comment:
```suggestion
		t.Fatalf("expected router deployment to have ROUTER_THREADS env var set to %s: %v", defaultThreadsStr, err)
```

Reply:
Good catch! These have been fixed.

Reply:
The router deployment should always have some value for `ROUTER_THREADS` set, right?
The new error messages imply that there are cases where the router deployment _should not_ have any corresponding value for `ROUTER_THREADS`.

Reply:
Correct, `ROUTER_THREADS` should always be set, even if `spec.tuningOptions.threadCount` isn't. Let me rework the error message for those cases

Reply:
Updated